File: app/src/main/java/hr/kacan/trznica/viewmodel/RegisterViewModel.java
package hr.kacan.trznica.viewmodel;

import android.app.Application;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import hr.kacan.trznica.models.Korisnik;
import hr.kacan.trznica.models.ResponseKorisnik;
import hr.kacan.trznica.repository.RegisterRepository;


public class RegisterViewModel extends AndroidViewModel {


    private MutableLiveData<ResponseKorisnik> mutableLiveData;
    private RegisterRepository registerRepository;


    private Korisnik korisnik;

    public Korisnik getKorisnik() {
        return korisnik;
    }

    public void setKorisnik(Korisnik korisnik) {
        this.korisnik = korisnik;
    }

    public RegisterViewModel(Application application) {
        super(application);
        registerRepository = RegisterRepository.getInstance();
    }

    public LiveData<ResponseKorisnik> registerKorisnik(Korisnik korisnik) {
        mutableLiveData = registerRepository.registerKorisnik(korisnik);
        return mutableLiveData;
    }

    public void getKorisnici() {

        //ploviloDAO.dodajNovoPlovilo(plovilo);
    }

    public LiveData<ResponseKorisnik> editKorisnik(Korisnik korisnik) {
        mutableLiveData = registerRepository.updateKorisnik(korisnik);
        return mutableLiveData;
    }

    public void deleteKorisnik() {

        //ploviloDAO.obrisiPlovilo(plovilo);
    }

}

File: app/src/main/java/hr/kacan/trznica/viewmodel/RegisterViewModel.kt
package hr.kacan.trznica.viewmodel

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import hr.kacan.trznica.models.Korisnik
import hr.kacan.trznica.models.ResponseKorisnik
import hr.kacan.trznica.repository.RegisterRepository

class RegisterViewModel(application: Application) : AndroidViewModel(application) {

    private lateinit var mutableLiveData: MutableLiveData<ResponseKorisnik>
    private val registerRepository: RegisterRepository = RegisterRepository()

    fun registerKorisnik(korisnik: Korisnik): LiveData<ResponseKorisnik> {
        mutableLiveData = registerRepository.registerKorisnik(korisnik)
        return mutableLiveData
    }

    fun editKorisnik(korisnik: Korisnik): LiveData<ResponseKorisnik> {
        mutableLiveData = registerRepository.updateKorisnik(korisnik)
        return mutableLiveData
    }

}